app/actions/nodeNetworkActions.js

```diff
@@ -10,67 +11,58 @@ import {
 } from '../core/constants'
 
 const ID = 'nodeNetwork'
+const RPC_PING_OVERRIDE = 5000
 
-const getBlockCount = async node =>
-  new Promise(resolve => {
-    let url = node.protocol ? `${node.protocol}://${node.url}` : node.url
-    url = node.port ? `${url}:${node.port}` : url
+const pingNodes = (nodes: Array<any>) => {
+  settings.timeout.ping = RPC_PING_OVERRIDE
+  const responses = []
+  return new Promise(resolve => {
+    nodes.forEach(node => {
+      let url = node.protocol ? `${node.protocol}://${node.url}` : node.url
+      url = node.port ? `${url}:${node.port}` : url
 
-    const client = new rpc.RPCClient(url)
-    client
-      .ping()
-      .then(latency => {
-        if (client.lastSeenHeight !== 0) {
-          resolve({
-            url,
-            blockCount: client.lastSeenHeight,
-            latency,
-          })
-        }
-      })
-      .catch({})
-  })
+      const maxWaitTimer = setTimeout(() => {
+        resolve(responses)
+      }, RPC_PING_OVERRIDE)
 
-const raceNodePromises = (total, promises) => {
-  const responses = []
-  return new Promise(resolve =>
-    promises.forEach(promise =>
-      promise.then(result => {
-        responses.push(result)
-        if (responses.length === total) resolve(responses)
-      }),
-    ),
-  )
+      const client = new rpc.RPCClient(url)
+      client
+        .ping()
+        .then(latency => {
+          if (client.lastSeenHeight !== 0) {
+            responses.push({
+              url,
+              blockCount: client.lastSeenHeight,
+              latency,
+            })
+          }
+          if (responses.length === nodes.length) {
+            clearTimeout(maxWaitTimer)
+            resolve(responses)
+          }
+        })
+        .catch(console.error)
+    })
+  })
```

Comment:
I think the timer method could be optimized with some nifty [Promise enhancements](https://stackoverflow.com/questions/48577702/setting-a-timeout-for-each-promise-within-a-promise-all).
Something like this:

Utils:
```js
Promise.delay = function(t) {
    return new Promise(resolve => {
        setTimeout(resolve, t);
    });
}

Promise.raceAll = function(promises, timeoutTime) {
    return Promise.all(promises.map(p => {
        return Promise.race([p, Promise.delay(timeoutTime)])
    }));
}
```
Usage:
```js
const pingNode = (node) => {
  return new Promise(resolve => {
    let url = node.protocol ? `${node.protocol}://${node.url}` : node.url
    url = node.port ? `${url}:${node.port}` : url
    new rpc.RPCClient(url)
      .ping()
      .then(latency => {
        if (client.lastSeenHeight !== 0) {
          resolve({
            url,
            blockCount: client.lastSeenHeight,
            latency,
          })
        }
      })
  })
}

const pingNodes = (nodes: Array<any>) => {
  settings.timeout.ping = RPC_PING_OVERRIDE
  return Promise.raceAll(nodes.map(pingNode), RPC_PING_OVERRIDE)
}
```
Why better?
* Doesn't rely on individual node ping promises to eventually resolve the accumulated responses.
* Doesn't recreate `maxWaitTimer` for each node ping promise.

Reply:
Wow I really like this strategy 🙌 took my a minute to wrap my head around but is much cleaner than is what is currently in the PR. Implementing now 😎

Reply:
Yay! Basically, there each node has a race between the timeout and the ping. The timeout rejects, the ping resolves. Whichever resolves gets added to the eventual response array.

```js
pingNodes(nodes).then((nodesThatPinged: Array<Node>) => {
  ...
});
```